docs/reference/modules/discovery/discovery-settings.asciidoc

```diff
@@ -199,6 +199,15 @@ Sets how long the master node waits for each cluster state update to be
 completely published to all nodes, unless `discovery.type` is set to
 `single-node`. The default value is `30s`. See <<cluster-state-publishing>>.
 
+`cluster.join_validation.cache_timeout`::
+(<<static-cluster-setting,Static>>)
```

Comment:
Do we actually want to document this?

This seems like a really tricky implementation detail for one. Also, maybe we will evolve this functionality further and it becomes meaningless in the mid term and then we have to deal with the whole deprecation noise? :)

Reply:
We make it a setting just in case there's some unforeseen problem that can be fixed by changing it, and IMO it's a bit rubbish when a problem can be fixed by adjusting a setting that isn't documented. We quite rightly get asked to add docs for such settings later. Note also that this whole section of the docs does have the following disclaimer:

> **WARNING**: If you adjust these settings then your cluster may not form correctly or may become unstable or intolerant of certain failures.

We're ok with making settings become no-ops if the implementation moves on, like e.g. we did with `cluster.join.timeout` in 7.x. That's not a breaking change.